File: xpath/xpath20/Java/XPath20ParserBase.java
import org.antlr.v4.runtime.*;

public abstract class XPath20ParserBase extends Parser
{
    protected XPath20ParserBase(TokenStream input)
    {
        super(input);
    }

    protected boolean IsFuncCall()
    {
		return !(
		    getInputStream().LA(1) == XPath20Parser.KW_ARRAY
		    || getInputStream().LA(1) == XPath20Parser.KW_ATTRIBUTE
			|| getInputStream().LA(1) == XPath20Parser.KW_COMMENT
			|| getInputStream().LA(1) == XPath20Parser.KW_DOCUMENT_NODE
			|| getInputStream().LA(1) == XPath20Parser.KW_ELEMENT
			|| getInputStream().LA(1) == XPath20Parser.KW_EMPTY_SEQUENCE
			|| getInputStream().LA(1) == XPath20Parser.KW_FUNCTION
			|| getInputStream().LA(1) == XPath20Parser.KW_IF
			|| getInputStream().LA(1) == XPath20Parser.KW_ITEM
			|| getInputStream().LA(1) == XPath20Parser.KW_MAP
			|| getInputStream().LA(1) == XPath20Parser.KW_NAMESPACE_NODE
			|| getInputStream().LA(1) == XPath20Parser.KW_NODE
			|| getInputStream().LA(1) == XPath20Parser.KW_PROCESSING_INSTRUCTION
			|| getInputStream().LA(1) == XPath20Parser.KW_SCHEMA_ATTRIBUTE
			|| getInputStream().LA(1) == XPath20Parser.KW_SCHEMA_ELEMENT
			|| getInputStream().LA(1) == XPath20Parser.KW_TEXT
		);
    }
}

File: xpath/xpath20/JavaScript/XPath20ParserBase.js
import antlr4 from 'antlr4';
import XPath20Parser from './XPath20Parser.js';

export default class XPath20ParserBase extends antlr4.Parser {

    constructor(input) {
        super(input);
    }

    IsFuncCall() {
        const nextTokenType = this._input.LT(1).type;
        return !(
		 nextTokenType == this.KW_ARRAY
		 || nextTokenType == this.KW_ATTRIBUTE
		 || nextTokenType == this.KW_COMMENT
		 || nextTokenType == this.KW_DOCUMENT_NODE
		 || nextTokenType == this.KW_ELEMENT
		 || nextTokenType == this.KW_EMPTY_SEQUENCE
		 || nextTokenType == this.KW_FUNCTION
		 || nextTokenType == this.KW_IF
		 || nextTokenType == this.KW_ITEM
		 || nextTokenType == this.KW_MAP
		 || nextTokenType == this.KW_NAMESPACE_NODE
		 || nextTokenType == this.KW_NODE
		 || nextTokenType == this.KW_PROCESSING_INSTRUCTION
		 || nextTokenType == this.KW_SCHEMA_ATTRIBUTE
		 || nextTokenType == this.KW_SCHEMA_ELEMENT
		 || nextTokenType == this.KW_TEXT
               );
    }
}
